pip/qsharp/noisy_simulator/_noisy_simulator.pyi

```diff
@@ -0,0 +1,238 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+from typing import Optional, List
+
+class NoisySimulatorError(BaseException):
+    """
+    EXPERIMENTAL:
+
+    An error returned from the Q# noisy simulator.
+    """
+
+    ...
+
+class Operation:
+    """
+    EXPERIMENTAL:
+
+    This struct represents a quantum operation. A quantum operation is a linear
+    transformation that maps a valid density matrix to another valid density matrices.
+    """
+
+    def __init__(self, kraus_operators: List[List[List[complex]]]) -> None:
```

Comment:
As I mentioned in the meeting, these types give an error when passing a NumPy matrix, even though it does work. No need to fix this PR, but open an issue to fix in a follow up. Thanks.

Reply:
Thanks! Will open the issue after merging. I found this, seems there is hope of writing a correct type hint:

https://numpy.org/doc/stable/reference/typing.html#module-numpy.typing

Apparently you can just take a dependency on `numpy.typing` without taking a dependency on the entire `numpy` library.